# EDA Study Case 02 

Kelompok 12 

**Ketua Kelompok**
*   Raynaldi Ilham Nanda Nur Pratama - 11210940000055

Anggota 
*   Heny Nur Afifah - 11210940000024
*   Noval Firdaus Ramadhan - 11210940000043
*   Syifa Ghaisani - 11210940000050
*   Athaya Talitha Hasna - 11210940000058
*   M Aqlani Wafi - 11210940000070
*   Selvi Oktaviani	- 11210940000081

In [37]:
import warnings 
warnings.filterwarnings('ignore')
import pandas as pd
import numpy as np

In [38]:
# Running Locally, yakinkan "file_" berada di folder "data"

try: 
    mengajar = pd.read_csv("data/mengajar.csv", low_memory = False, encoding='utf8')
    dosen = pd.read_csv("data/tb_dosen.csv", low_memory = False, encoding='utf8')
    kelas = pd.read_csv("data/tb_kelas.csv", low_memory = False, encoding='utf8')
    krs = pd.read_csv("data/tb_krs.csv", low_memory = False, encoding='utf8')
    mahasiswa = pd.read_csv("data/tb_mahasiswa.csv", low_memory = False, encoding='utf8')
    kuliah = pd.read_csv("data/tb_mata_kuliah.csv", low_memory = False, encoding='utf8')
    nilai = pd.read_csv("data/tb_nilai.csv", low_memory = False, encoding='utf8')
except:
    !mkdir data
    !wget -P data/ https://raw.githubusercontent.com/Raynaldiilhamnp/EDA-studi-kasus-02-kelompok-12/main/data/mengajar.csv
    !wget -P data/ https://raw.githubusercontent.com/Raynaldiilhamnp/EDA-studi-kasus-02-kelompok-12/main/data/tb_dosen.csv
    !wget -P data/ https://raw.githubusercontent.com/Raynaldiilhamnp/EDA-studi-kasus-02-kelompok-12/main/data/tb_kelas.csv
    !wget -P data/ https://raw.githubusercontent.com/Raynaldiilhamnp/EDA-studi-kasus-02-kelompok-12/main/data/tb_krs.csv
    !wget -P data/ https://raw.githubusercontent.com/Raynaldiilhamnp/EDA-studi-kasus-02-kelompok-12/main/data/tb_mahasiswa.csv
    !wget -P data/ https://raw.githubusercontent.com/Raynaldiilhamnp/EDA-studi-kasus-02-kelompok-12/main/data/tb_mata_kuliah.csv
    !wget -P data/ https://raw.githubusercontent.com/Raynaldiilhamnp/EDA-studi-kasus-02-kelompok-12/main/data/tb_nilai.csv
    mengajar = pd.read_csv("data/mengajar.csv", low_memory = False, encoding='utf8')
    dosen = pd.read_csv("data/tb_dosen.csv", low_memory = False, encoding='utf8')
    kelas = pd.read_csv("data/tb_kelas.csv", low_memory = False, encoding='utf8')
    krs = pd.read_csv("data/tb_krs.csv", low_memory = False, encoding='utf8')
    mahasiswa = pd.read_csv("data/tb_mahasiswa.csv", low_memory = False, encoding='utf8')
    kuliah = pd.read_csv("data/tb_mata_kuliah.csv", low_memory = False, encoding='utf8')
    nilai = pd.read_csv("data/tb_nilai.csv", low_memory = False, encoding='utf8')

mengajar.shape, dosen.shape, kelas.shape, krs.shape, mahasiswa.shape, kuliah.shape, nilai.shape

((14, 4), (4, 6), (4, 4), (7, 6), (15, 8), (9, 3), (33, 7))

# **DATA**



In [39]:
# Untuk melihat data bagian tabel mengajar
mengajar

NIDN kode_MK Semester  Tahun
0    123   KAL01   ganjil   2020
1    123   KAL01   ganjil   2019
2    123   STA01    genap   2020
3    123   MAT01    genap   2020
4    123   STA02   ganjil   2019
5    234    OR01   ganjil   2019
6    234    CS01    genap   2020
7    787   KAL01   ganjil   2021
8    787   KAL01   ganjil   2022
9    787   STA02   ganjil   2021
10   345   MAT73    genap   2020
11   345   MAT53    genap   2019
12   345   STA02   ganjil   2019
13   345   KAL02   ganjil   2020

In [40]:
# Untuk melihat data bagian tabel dosen
dosen

NIDN           nama             email Pendidikan_Terakhir  Telephone  \
0   123   Andri Basuki   anbr@googol.com                  S3   11230000   
1   234       zulfikar   dosen@email.com                  S3   22340000   
2   787           yudi    yudi@email.com                  S2   55670000   
3   345  Annisa Fitria  annniza@yyyy.com                  S2   88970000   

  Jenis_Kelamin  
0          Pria  
1          Pria  
2          Pria  
3        Wanita

In [41]:
# Untuk melihat data bagian tabel kelas
kelas
# Variabel yang tidak digunakan

kode_kelas nama_kelas lokasi_kelas  Kapasitas
0      SIS1J  Reguler A     Gedung D         10
1       D101  Reguler A     Gedung D         15
2       D212  Reguler B     Gedung D         40
3       P601  Reguler B   Gedung PLT         30

In [42]:
# Untuk melihat data bagian tabel krs
krs
# Variabel yang tidak digunakan

id id_mata_kuliah kode_kelas     nim  tahun semester
0   1          KAL01      SIS1J  193001   2021    genap
1   2          STA01       D101  193001   2021    genap
2   3           OR01       D212  193001   2021    genap
3   4          MK002       P601  194001   2021    genap
4   5          MK002       P601  194001   2021    genap
5   6          MK003      SIS1J  194001   2021    genap
6   7          MK004       D101  194001   2021    genap

In [43]:
# Untuk melihat data bagian tabel mahasiswa
mahasiswa

nim            nama                    email  Gender Asal_Sekolah  \
0   194015   Muhammad Amir       Samarinda@inet.com    Pria          SMU   
1   194001   Samsul Kurnia        Bontang@email.com    Pria          SMU   
2   194002  Bambang Arifin         Sangata@inet.com    Pria          SMU   
3   194003            Udin            Udin@inet.com    Pria           MA   
4   194004            Iwan            Iwan@inet.com    Pria           MA   
5   194005  Bambang Arifin  Bambang Arifin@inet.com    Pria          SMK   
6   194006            Wati            Wati@inet.com  Wanita          SMK   
7   194007             Ani             Ani@inet.com  Wanita          SMK   
8   194008       Rahmawati       Rahmawati@inet.com  Wanita          SMU   
9   194009            Asep            Asep@inet.com    Pria          SMU   
10  194010            Rina            rina@inet.com  Wanita           MA   
11  194011            Sari            sari@inet.com  Wanita           MA   
12  194012            Bowo            Bowo@inet.com    Pria          SMU   
13  194013            Budi            Budi@inet.com    Pria          SMK   
14  194014           Putri           putri@inet.com  Wanita          SMK   

    Telephone  Angkatan  Asal_Daerah  
0    12345678      2019  Luar Daerah  
1    12345679      2019        Lokal  
2    12345680      2019  Luar Daerah  
3    12345681      2019        Lokal  
4    12345682      2019  Luar Daerah  
5    12345683      2019        Lokal  
6    12345684      2019        Lokal  
7    12345685      2019  Luar Daerah  
8    12345686      2020        Lokal  
9    12345687      2020  Luar Daerah  
10   12345688      2020        Lokal  
11   12345689      2020        Lokal  
12   12345690      2020  Luar Daerah  
13   12345691      2020        Lokal  
14   12345692      2020        Lokal

In [44]:
# Untuk melihat data bagian tabel mata kuliah
kuliah

id_mata_kuliah           nama_mata_kuliah  SKS
0          KAL01             Kalkulus Dasar    3
1          STA01       Statistika Elementer    3
2          MAT01           Matematika Dasar    3
3          STA02      Statistika Matematika    4
4           OR01         Operation Research    4
5           CS01  Pendahuluan Ilmu Komputer    3
6          MAT73            Aljabar Abstrak    4
7          MAT53             Aljabar linear    3
8          KAL02            Kalkulus Lanjut    4

In [45]:
# Untuk melihat data bagian tabel nilai
nilai

nim id_mata_kuliah  NIDN  nilai Semester  Tahun     rand
0   194015          KAL01   123     70   ganjil   2019  0.50463
1   194001          KAL01   123     80   ganjil   2019      NaN
2   194002          KAL01   123     85   ganjil   2019      NaN
3   194003          KAL01   123     90   ganjil   2019      NaN
4   194004          KAL01   123     92   ganjil   2019      NaN
5   194005          KAL01   123     91   ganjil   2019      NaN
6   194006          KAL01   123     93   ganjil   2019      NaN
7   194007          KAL01   123     85   ganjil   2019      NaN
8   194008          KAL01   787     60   ganjil   2020      NaN
9   194009          KAL01   787     63   ganjil   2020      NaN
10  194010          KAL01   787     62   ganjil   2020      NaN
11  194011          KAL01   787     61   ganjil   2020      NaN
12  194012          KAL01   787     70   ganjil   2020      NaN
13  194013          KAL01   787     90   ganjil   2020      NaN
14  194014          KAL01   787     88   ganjil   2020      NaN
15  194004           CS01   234     90    genap   2020      NaN
16  194005           CS01   234     92    genap   2020      NaN
17  194006           CS01   234     85    genap   2020      NaN
18  194007           CS01   234     55    genap   2020      NaN
19  194008           CS01   234     60    genap   2020      NaN
20  194009           CS01   234     91    genap   2020      NaN
21  194010           CS01   234     90    genap   2020      NaN
22  194015          STA02   345     40   ganjil   2021      NaN
23  194002          STA02   345     50   ganjil   2021      NaN
24  194004          STA02   345     56   ganjil   2021      NaN
25  194007          STA02   345     70   ganjil   2021      NaN
26  194008          STA02   345     72   ganjil   2021      NaN
27  194011          STA02   345     31   ganjil   2021      NaN
28  194014          STA02   345     88   ganjil   2021      NaN
29  194001          KAL02   345     55   ganjil   2020      NaN
30  194002          KAL02   345     60   ganjil   2020      NaN
31  194004          KAL02   345     65   ganjil   2020      NaN
32  194007          KAL02   345     74   ganjil   2020      NaN

# **Nomor 1**

Apakah ada dosen yang cenderung memberikan nilai cenderung lebih tinggi (atau lebih rendah) dibandingkan dosen lainnya?

In [46]:
nilai_dosen = nilai.groupby(nilai['NIDN']).mean()[['nilai']]
nilai_dosen = nilai_dosen.rename({'nilai': 'mean_nilai'}, axis=1) 
nilai_dosen = pd.merge(dosen, nilai_dosen, left_on='NIDN', right_index=True)
nilai_dosen = nilai_dosen[['NIDN', 'nama', 'mean_nilai']]
sb = nilai.groupby(nilai['NIDN']).std()[['nilai']]
sb = sb.rename({'nilai': 'std_nilai'}, axis=1) 
nilai_dosen = nilai_dosen.set_index(['NIDN']) 
nilai_dosen = pd.concat([nilai_dosen, sb], axis=1)
nilai_dosen

nama  mean_nilai  std_nilai
NIDN                                      
123    Andri Basuki   85.750000   7.741355
234        zulfikar   80.428571  15.883504
345   Annisa Fitria   60.090909  16.220077
787            yudi   70.571429  13.010984

In [47]:
a = nilai.merge(dosen, how="outer")
a = a[['NIDN','nama','id_mata_kuliah', 'nilai']]
a = a.groupby([a['NIDN'], a['nama'], a['id_mata_kuliah']]).mean()[['nilai']]
a = a.rename({'nilai': 'mean_nilai'}, axis=1) 
a
b = nilai.merge(dosen, how="outer")
b = b[['NIDN','nama','id_mata_kuliah', 'nilai']]
b = b.groupby([b['NIDN'], b['nama'], b['id_mata_kuliah']]).std()[['nilai']]
b = b.rename({'nilai': 'std_nilai'}, axis=1)
b = pd.concat([a, b], axis=1) 
b

mean_nilai  std_nilai
NIDN nama          id_mata_kuliah                       
123  Andri Basuki  KAL01            85.750000   7.741355
234  zulfikar      CS01             80.428571  15.883504
345  Annisa Fitria KAL02            63.500000   8.103497
                   STA02            58.142857  19.836234
787  yudi          KAL01            70.571429  13.010984

**Interpretasi no. 1**

Berdasarkan analisis yang telah kami lakukan bahwa terdapat dosen yang cenderung memberikan nilai lebih tinggi dan cenderung lebih rendah dibandingkan dosen lainnya. Dosen yang cenderung memberikan nilai yang lebih tinggi yaitu bapak Andri Basuki. Beliau memberikan nilai yang lebih tinggi pada mata kuliah Kalkulus 1. Berdasarkan analisis data diketahui bahwa bapak Andri Basuki memberikan rata rata nilai sebesar 85.75 dengan standar deviasi sebesar 7.741355.

Sedangkan dosen yang cenderung memberikan nilai lebih rendah yaitu Ibu Annisa Fitria, beliau mengajar dua mata kuliah yaitu Kalkulus 2 dan Statistika 2. Berdasarkan analisis data diketahui bahwa Ibu Annisa Fitria memberikan nilai yang cenderung lebih rendah pada mata kuliah Statistika 2 dengan rata rata sebesar 58.142857 dan dengan standar deviasi sebesar 19.836234. Meskipun ibu Annisa Fitria mengajar dua mata kuliah namun semua nilai yang diberikan oleh beliau cenderung lebih rendah dibandingkan dengan dosen lainnya.

Namun, analisis ini kurang akurat dikarenakan adanya beberapa faktor, diantaranya:

1. Perbedaan jumlah kelas yang diajar oleh masing-masing dosen
2. Perbedaan persentase penilaian dalam mengajar yang ditetapkan oleh tiap dosen.

Bapak Andri Basuki hanya mengajar pada kelas kalkulus1 yang tentu jumlah mahasiswanya sedikit sehingga membuat rata rata nilai mahasiswanya menjadi lebih tinggi diantara dosen lain. Data ini belum sepenuhnya akurat karena beberapa variabel yang dibutuh tidak ada pada data, seperti persentasi penilaian dosen yang tentu saja berpengaruh pada nilai mahasiswa, ada kemungkinan mahasiswa yang diajar bapak andri memang memperhatikan persentase penilai dosen. Atau ada kemungkinan lain bapak Andri Basuki memang dosen yang memberikan nilai dengan baik jadi nilai mahasiswa yang diajarnya lebih tinggi dari dosen lain, tetapi pernyataan ini harus dibuktikan dengan pendapat mahasiswa bagaimana bapak andri memberikan nilai.

Ibu Anisa Fitria mengajar pada dua mata kuliah, tentu jumlah mahasiswa yang diajar lebih banyak sehingga menyebabkan rata rata nilai mahasiswa yang diajar lebih rendah dibandingkan dosen lain. Fakta data lainnya mengenai bu Anisa Fitria pada kelas mata kuliah kalkulus dua memiliki rata2 paling rendah dengan standar deviasi kecil, berarti pada kelas tersebut memang terbukti nilai paling rendah, ada beberapa kemungkinan pada kasus ini, apakah mata kuliahnya yang sulit atau memang terbukti bu Anisa Fitria memang dosen yang memberikan nilai paling rendah, untuk meyakinkan fakta yang menguatkan lagi kita memerlukan survey mahasiswa, manakah kemungkinan yang paling benar.

# **Nomor 2**

Menurut data yang ada, apakah terdapat perbedaan dalam pemberian nilai antara dosen dengan Pendidikan Terakhir S3 dan S2?

In [48]:
df_penilaian_dosen = pd.merge(nilai, dosen, how="outer")
penilaian_dosen = df_penilaian_dosen[['NIDN', 'nama',  'nilai','Pendidikan_Terakhir', 'id_mata_kuliah']]
penilaian_dosen = penilaian_dosen.groupby([penilaian_dosen['Pendidikan_Terakhir'], penilaian_dosen['id_mata_kuliah']]).mean()[['nilai']]
penilaian_dosen = penilaian_dosen.rename({'nilai': 'mean_nilai'}, axis=1)

std = nilai.groupby(nilai['NIDN']).std()[['nilai']]
std = std.rename({'nilai': 'std_nilai'}, axis=1) 

std_nilai_dosen = df_penilaian_dosen[['NIDN', 'nama',  'nilai','Pendidikan_Terakhir', 'id_mata_kuliah']]
std_nilai_dosen = std_nilai_dosen.groupby([std_nilai_dosen['Pendidikan_Terakhir'], std_nilai_dosen['id_mata_kuliah']]).std()[['nilai']]
std_nilai_dosen = std_nilai_dosen.rename({'nilai': 'std_nilai'}, axis=1)
penilaian_dosen = pd.concat([penilaian_dosen, std_nilai_dosen], axis=1)
penilaian_dosen

mean_nilai  std_nilai
Pendidikan_Terakhir id_mata_kuliah                       
S2                  KAL01            70.571429  13.010984
                    KAL02            63.500000   8.103497
                    STA02            58.142857  19.836234
S3                  CS01             80.428571  15.883504
                    KAL01            85.750000   7.741355

**Interpretasi No. 2 :**

Berdasarkan data tabel diatas, terlihat bahwa terdapat perbedaan dalam pemberian nilai antara dosen dengan pendidikan terakhir S3 dan dosen dengan pendidikan terakhir S2.

Dosen S3 yaitu Pak Andri Basuki dan Pak Zulfikar cenderung memberikan nilai yang lebih tinggi dibandingkan dengan dosen S2 yaitu Bu Annisa Fitria dan Pak Yudi. Nilai standar deviasi untuk dosen S3 cenderung lebih rendah sehingga data ini baik dan dapat memperkuat bahwa dosen berpendidikan terakhir S3 memang memberikan nilai yang lebih tinggi dibandingkan dosen yang berpendidikan terakhir S2.

Berdasarkan analisis yang telah kami lakukan bahwa terdapat perbedaan pemberian nilai antara dosen dengan pendidikan terakhir S3 dan dosen dengan pendidikan terakhir S2. Dosen dengan pendidikan akhir S3 memberikan penilaian yang lebih baik bila dibandingkan dengan dosen berpendidikan akhir S2.

Terdapat dua dosen yang memberikan nilai cenderung lebih tinggi dengan pendidikan terakhir S3 yaitu:

Pak Andri Basuki dengan pemberian rata-rata nilai 85.75 dan standard deviasi 7.741355 pada mata kuliah Kalkulus 1;
Pak Zulfikar dengan pemberian rata-rata nilai 80.428571 dan standar deviasi 15.883504 pada mata kuliah Pendahuluan Ilmu Komputer.
Lalu terdapat dua dosen yang memberikan nilai cenderung lebih rendah dengan pendidikan terakhir S2, salah satu dosen mengajar dua mata kuliah yaitu:

Pak Yadi dengan pemberian rata-rata nilai 70.571429 dan standar deviasi 13.010984 pada mata kuliah Kalkulus 1;
Bu Annisa Fitria dengan pemberian rata-rata nilai 63.5 dan standar deviasi 8.103497 pada mata kuliah Kalkulus 2;
Bu Annisa Fitria dengan pemberian rata-rata nilai 58.142857 dan standar deviasi 19.836234 pada mata kuliah Statistika 2.

# **Nomor 3**

Apakah terdapat perbedaan performa akademik antara mahasiswa yang berasal dari Madrasah/SMK dan SMU?

In [49]:
y = pd.merge(nilai, mahasiswa, how = 'outer')
new2 = y[['nilai', 'Asal_Daerah', 'Asal_Sekolah', 'id_mata_kuliah']]
Mahasiswa = new2.groupby([new2['Asal_Daerah'], new2['Asal_Sekolah']]).mean()[['nilai']]
Mahasiswa = Mahasiswa.rename({'nilai': 'mean_nilai'}, axis=1)
# Mahasiswa
Mahasiswa2 = new2.groupby([new2['Asal_Daerah'], new2['Asal_Sekolah']]).std()[['nilai']]
Mahasiswa2 = Mahasiswa2.rename({'nilai': 'std_nilai'}, axis=1)
Mahasiswa2 = pd.concat([Mahasiswa, Mahasiswa2], axis=1) 
Mahasiswa2


mean_nilai  std_nilai
Asal_Daerah Asal_Sekolah                       
Lokal       MA             66.800000  24.570307
            SMK            89.571429   2.760262
            SMU            65.400000  10.285913
Luar Daerah MA             75.750000  18.006943
            SMK            71.000000  12.409674
            SMU            66.125000  16.881414

**Interpretasi No. 3 :**

Berdasarkan analisis yang telah kami lakukan bahwa terdapat perbedaan performa akademik antara mahasiswa yang berasal dari Madrasah, SMK dan SMU. Pada ananlisis kali ini kami juga menambah faktor lainnya yaitu daerah asal mahasiswa, yaitu lokal dan luar daerah.

Lokal

lulusan MA memiliki rata-rata nilai 66.8 dan standar deviasi 24.570307
lulusan SMK memiliki rata-rata nilai 89.571429 dan standar deviasi 2.760262
lulusan SMU memiliki rata-rata nilai 65.4 dan standar deviasi 10.285913
Luar Daerah

lulusan MA memiliki rata-rata nilai 75.75 dan standar deviasi 18.006943
lulusan SMK memiliki rata-rata nilai 71 dan standar deviasi 12.409674
lulusan SMU memiliki rata-rata nilai 66.125 dan standar deviasi 16.881414
Performa akademik antara mahasiswa yang berasal dari MA, SMK, dan SMU, Berdasarkan analisis yang telah dilakukan didapatkan hasil bahwa mahasiswa yang berasal dari daerah lokal, SMK memiliki rata rata nilai paling tinggi dan standar deviasi paling rendah berarti nilai mereka rata rata memang merata tinggi, sedangkan pada luar daerah MA memiliki nilai rata rata tinggi dengan standar deviasi yang juga rendah. Tetapi secara keseluruhan performa paling baik pada SMK yang berada pada daerah lokal, karena beberapa faktor yang mendukung mereka seperti fasilitas yang memadai yang mereka dapatkan pada saat SMK dan mereka sudah terbiasa dengan penjurusan dan praktek pada saat menempuh pendidikan SMK, sehingga pengetahuan dasar akan dunia akademik saat kuliah bisa terbantu oleh hal tersebut.

Pada penilaian perbedaan performa akademik mahasiswa, masih kurang data untuk akurasi data, apakah data ini bisa dikatakan benar, karena dia hanya berfokus pada nilai mahasiswa. Berbedaan background pendidikan tentu saja mempengaruhi kepahaman mahasiswa, serta pola pikir mereka dan tentu saja dosen memberikan materi perkuliahan dalam kapasitas yang sama tanpa melihat background mahasiswanya. Selain itu penjurusan mereka saat pendidikan sebelumnya juga bisa mempengaruhi performa mereka saat perkuliahan, tetapi tidak terdapat data yang menjelaskan hal tersebut.

Berdasarkan tabel tersebut, nilai rata-rata mahasiswa yang berasal dari SMK memiliki nilai yang tinggi dibandingkan nilai rata-rata mahasiswa dari SMU dan Madrasah. Oleh karena itu, dapat disimpulkan bahwa mahasiswa yang berasal dari SMK memiliki nilai yang lebih tinggi. Sehingga mahasiswa SMK memiliki performa akademik yang lebih unggul dibandingkan dengan mahasiswa yang berasal dari SMU dan Madrasah. Artinya terdapat perbedaan performa akademik antara mahasiswa yang berasal dari Madrasah, SMK, dan SMU.

# **Nomor 4**

Apakah ada perbedaan yang nampak di data terhadap kemampuan akademik antara mahasiswa dan mahasiswi?

In [50]:
mahasiswa

nim            nama                    email  Gender Asal_Sekolah  \
0   194015   Muhammad Amir       Samarinda@inet.com    Pria          SMU   
1   194001   Samsul Kurnia        Bontang@email.com    Pria          SMU   
2   194002  Bambang Arifin         Sangata@inet.com    Pria          SMU   
3   194003            Udin            Udin@inet.com    Pria           MA   
4   194004            Iwan            Iwan@inet.com    Pria           MA   
5   194005  Bambang Arifin  Bambang Arifin@inet.com    Pria          SMK   
6   194006            Wati            Wati@inet.com  Wanita          SMK   
7   194007             Ani             Ani@inet.com  Wanita          SMK   
8   194008       Rahmawati       Rahmawati@inet.com  Wanita          SMU   
9   194009            Asep            Asep@inet.com    Pria          SMU   
10  194010            Rina            rina@inet.com  Wanita           MA   
11  194011            Sari            sari@inet.com  Wanita           MA   
12  194012            Bowo            Bowo@inet.com    Pria          SMU   
13  194013            Budi            Budi@inet.com    Pria          SMK   
14  194014           Putri           putri@inet.com  Wanita          SMK   

    Telephone  Angkatan  Asal_Daerah  
0    12345678      2019  Luar Daerah  
1    12345679      2019        Lokal  
2    12345680      2019  Luar Daerah  
3    12345681      2019        Lokal  
4    12345682      2019  Luar Daerah  
5    12345683      2019        Lokal  
6    12345684      2019        Lokal  
7    12345685      2019  Luar Daerah  
8    12345686      2020        Lokal  
9    12345687      2020  Luar Daerah  
10   12345688      2020        Lokal  
11   12345689      2020        Lokal  
12   12345690      2020  Luar Daerah  
13   12345691      2020        Lokal  
14   12345692      2020        Lokal

In [51]:
kemampuan_gender = pd.merge(mahasiswa, nilai, how='outer')
kemampuan_gender = kemampuan_gender[['nim', 'Gender', 'Tahun', 'nilai', 'id_mata_kuliah', 'Angkatan']]
kemampuan_gender = kemampuan_gender.groupby([kemampuan_gender['Angkatan'],kemampuan_gender['Tahun'], kemampuan_gender['Gender'], kemampuan_gender['id_mata_kuliah']]).mean()[['nilai']]
kemampuan_gender = kemampuan_gender.rename({'nilai': 'mean_nilai'}, axis=1) 
kemampuan_gender = kemampuan_gender.astype(int)
kemampuan_gender
# std_kemampuan_gender = pd.merge(mahasiswa, nilai, how='outer')
# std_kemampuan_gender = std_kemampuan_gender[['nim', 'Gender', 'Tahun', 'nilai', 'id_mata_kuliah', 'Angkatan']]
# std_kemampuan_gender = std_kemampuan_gender.groupby([std_kemampuan_gender['Angkatan'],std_kemampuan_gender['Tahun'], std_kemampuan_gender['Gender'], std_kemampuan_gender['id_mata_kuliah']]).std()[['nilai']]
# std_kemampuan_gender = std_kemampuan_gender.rename({'nilai': 'std_nilai'}, axis=1) 
# # std_kemampuan_gender = std_kemampuan_gender.astype(int)
# std_kemampuan_gender

mean_nilai
Angkatan Tahun Gender id_mata_kuliah            
2019     2019  Pria   KAL01                   84
               Wanita KAL01                   89
         2020  Pria   CS01                    91
                      KAL02                   60
               Wanita CS01                    70
                      KAL02                   74
         2021  Pria   STA02                   48
               Wanita STA02                   70
2020     2020  Pria   CS01                    91
                      KAL01                   74
               Wanita CS01                    75
                      KAL01                   67
         2021  Wanita STA02                   63

**Interpretasi No. 4 :**

Berdasarkan hasil analisis yang telah kami lakukan, diketahui bahwa terdapat perbedaan nilai akademik antara mahasiswa dan mahasiswi. 

Untuk mata kuliah kalkulus 1 pada angkatan 2019 maka nilai mahasiswi lebih unggul 5 poin dibandingkan dengan nilai mahasiswanya, namun pada angkatan 2020 justru nilai mahasiswalah yang lebih unggul 7 poin dibandingkan dengan nilai mahasiswinya. 

Untuk mata kuliah Pendahuluan Ilmu Komputer pada angkatan 2019 dan 2020 diketahui bahwa nilai mahasiswa lebih unggul 20 poin lebih dibandingkan nilai mahasiswinya. 

Dari analisa diatas, disimpulkan bahwa mahasiswa memiliki nilai lebih unggul dalam bidang akademik dibandingkan dengan nilai mahasiswi. 

Namun, kesimpulan tersebut tidak akurat karena pada mata kuliah kalkulus 2 dari data diketahui hanya diambil oleh angkatan 2020 saja dengan nilai lebih unggul mahasiswi dibandingkan nilai mahasiswanya.  Dan untuk mata kuliah statistika 2 pada angkatan 2021 hanya diambil oleh mahasiswi tanpa ada data untuk nilai mahasiswanya.

# **Nomor 5**

Apakah ada perbedaan prestasi akademik yang nampak di data antar angkatan mahasiswa?

In [52]:
z = pd.merge(nilai, mahasiswa, how = 'outer')
z

nim id_mata_kuliah  NIDN  nilai Semester  Tahun     rand  \
0   194015          KAL01   123     70   ganjil   2019  0.50463   
1   194015          STA02   345     40   ganjil   2021      NaN   
2   194001          KAL01   123     80   ganjil   2019      NaN   
3   194001          KAL02   345     55   ganjil   2020      NaN   
4   194002          KAL01   123     85   ganjil   2019      NaN   
5   194002          STA02   345     50   ganjil   2021      NaN   
6   194002          KAL02   345     60   ganjil   2020      NaN   
7   194003          KAL01   123     90   ganjil   2019      NaN   
8   194004          KAL01   123     92   ganjil   2019      NaN   
9   194004           CS01   234     90    genap   2020      NaN   
10  194004          STA02   345     56   ganjil   2021      NaN   
11  194004          KAL02   345     65   ganjil   2020      NaN   
12  194005          KAL01   123     91   ganjil   2019      NaN   
13  194005           CS01   234     92    genap   2020      NaN   
14  194006          KAL01   123     93   ganjil   2019      NaN   
15  194006           CS01   234     85    genap   2020      NaN   
16  194007          KAL01   123     85   ganjil   2019      NaN   
17  194007           CS01   234     55    genap   2020      NaN   
18  194007          STA02   345     70   ganjil   2021      NaN   
19  194007          KAL02   345     74   ganjil   2020      NaN   
20  194008          KAL01   787     60   ganjil   2020      NaN   
21  194008           CS01   234     60    genap   2020      NaN   
22  194008          STA02   345     72   ganjil   2021      NaN   
23  194009          KAL01   787     63   ganjil   2020      NaN   
24  194009           CS01   234     91    genap   2020      NaN   
25  194010          KAL01   787     62   ganjil   2020      NaN   
26  194010           CS01   234     90    genap   2020      NaN   
27  194011          KAL01   787     61   ganjil   2020      NaN   
28  194011          STA02   345     31   ganjil   2021      NaN   
29  194012          KAL01   787     70   ganjil   2020      NaN   
30  194013          KAL01   787     90   ganjil   2020      NaN   
31  194014          KAL01   787     88   ganjil   2020      NaN   
32  194014          STA02   345     88   ganjil   2021      NaN   

              nama                    email  Gender Asal_Sekolah  Telephone  \
0    Muhammad Amir       Samarinda@inet.com    Pria          SMU   12345678   
1    Muhammad Amir       Samarinda@inet.com    Pria          SMU   12345678   
2    Samsul Kurnia        Bontang@email.com    Pria          SMU   12345679   
3    Samsul Kurnia        Bontang@email.com    Pria          SMU   12345679   
4   Bambang Arifin         Sangata@inet.com    Pria          SMU   12345680   
5   Bambang Arifin         Sangata@inet.com    Pria          SMU   12345680   
6   Bambang Arifin         Sangata@inet.com    Pria          SMU   12345680   
7             Udin            Udin@inet.com    Pria           MA   12345681   
8             Iwan            Iwan@inet.com    Pria           MA   12345682   
9             Iwan            Iwan@inet.com    Pria           MA   12345682   
10            Iwan            Iwan@inet.com    Pria           MA   12345682   
11            Iwan            Iwan@inet.com    Pria           MA   12345682   
12  Bambang Arifin  Bambang Arifin@inet.com    Pria          SMK   12345683   
13  Bambang Arifin  Bambang Arifin@inet.com    Pria          SMK   12345683   
14            Wati            Wati@inet.com  Wanita          SMK   12345684   
15            Wati            Wati@inet.com  Wanita          SMK   12345684   
16             Ani             Ani@inet.com  Wanita          SMK   12345685   
17             Ani             Ani@inet.com  Wanita          SMK   12345685   
18             Ani             Ani@inet.com  Wanita          SMK   12345685   
19             Ani             Ani@inet.com  Wanita          SMK   12345685   
20       Rahmawati       Rahmawati@inet.com  Wanita          SMU   12

In [53]:
new4 = z[['nilai', 'Angkatan']]
Angkatan = new4.groupby(new4['Angkatan']).mean()[['nilai']]
Angkatan = Angkatan.rename({'nilai': 'mean_nilai'}, axis=1)
Angkatan

mean_nilai
Angkatan            
2019       73.900000
2020       71.230769

**Interpretasi No. 5 :**

Berdasarkan tabel diatas menunjukkan bahwa nilai rata-rata angkatan 2019 lebih tinggi dibandingkan nilai rata-rata angkatan 2020. Sehingga dapat disimpulkan bahwa terdapat perbedaan prestasi akademik antar angkatan mahasiswa. Namun, perbedaan akademik tersebut tidak terlalu jauh karena selisih nilai rata-ratanya tidak terlalu besar.

# **Nomor 6**

Apakah ada permasalahan di data atau informasi lain yang anda temukan di data?

**Interpretasi No. 6 :**

Dari seluruh data yang ada, terdapat variabel rand pada tabel nilai yang didalamnya banyak NaN. Karena banyaknya NaN pada variabel rand, maka variabel ini tidak diperlukan dalam mengolah kasus ini. Terdapat beberapa tabel yang tidak digunakan yaitu tabel kelas yang berisi kode_kelas, nama_kelas dan lokasi_kelas. Selain itu ada terdapat juga beberapa variabel yang tidak terpakai, seperti  email, telephone, asal daerah, SKS dan semester. Data yang diberikan universitas masih banyak kekurangan, padahal sangat dibutuhkan dalam analisis kali ini, seperti  kriteria penilaian akademik yang masih kurang hanya berfokus pada nilai mahasiswa, terdapat kekurangan persentase penilaian dosen padahal itu sangat penting, dibutuhkan data tingkat pengetahuan dasar dari mahasiswa, data penjurusan pada pendidikan sebelum kuliah tidak ada, dan dibutuhkan data data akademik seperti menang perlombaan akademik dan nonakademik.

# **Kesimpulan**

Berdasarkan analisis yang telah kami lakukan,  terdapat dosen yang cenderung memberikan nilai lebih tinggi dan lebih rendah. Lalu terdapat dosen yang memberikan nilai cenderung lebih tinggi dan lebih rendah. Dosen yang memberikan nilai cenderung memberikan nilai lebih tinggi yaitu Bapak Andri Basuki, sedangkan dosen yang cenderung memberikan nilai lebih rendah yaitu Ibu Annisa Fitria. Lalu terdapat pula dalam pemberian nilai antara dosen dengan pendidikan akhir S3 yang memberikan penilaian lebih baik dibandingkan dengan dosen berpendidikan akhir S2. Lalu terdapat perbedaan performa akademik antara mahasiswa yang berasal dari madrasah, SMK, SMU dan didapatkan hasil bahwa mahasiswa lulusan SMK memiliki nilai lebih tinggi dibandingkan nilai rata-rata mahasiswa dari SMU dan Madrasah. Lalu terdapat perbedaan prestasi akademik antara angkatan 2019 dengan 2020, dengan hasil angakatan 2019 lebih tinggi dibandingkan angakatan 2020. Lalu terdapat banyak variabel yang tidak dibutuhkan, sedangkan data yang diperlukan masih banyak yang belum tersedia di data.